Source/PsDataPlugin/Classes/Collection/PsDataArrayProxy.h

```diff
@@ -155,6 +155,27 @@ struct FPsDataBaseArrayProxy
 		return Index;
 	}
 
+	template <typename PredicateType>
+	int32 RemoveAll(const PredicateType& Predicate)
+	{
+		static_assert(!bConst, "Unsupported method for FPsDataConstArrayProxy, use FPsDataArrayProxy");
+
+		int32 RemovedElements = 0;
+		TArray<T>& Array = Get();
+		for (auto It = Array.CreateIterator(); It; ++It)
+		{
+			if (Predicate(*It))
+			{
+				FDataReflectionTools::FArrayChangeBehavior<T>::RemoveFromArray(Instance.Get(), *It);
+				It.RemoveCurrent();
+				UPsDataEvent::DispatchChange(Instance.Get(), Field);
```

Comment:
Use `UPsDataEvent::DispatchChange(Instance.Get(), Field);` once after loop
For example:
```
 if (RemovedElements > 0)
{
 UPsDataEvent::DispatchChange(Instance.Get(), Field);
}
```